com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelController.java

```diff
@@ -27,55 +33,99 @@
 import com.eclipsesource.modelserver.emf.configuration.ServerConfiguration;
 import com.google.inject.Inject;
 
-import io.javalin.Context;
 import io.javalin.apibuilder.CrudHandler;
-import io.javalin.json.JavalinJackson;
+import io.javalin.http.BadRequestResponse;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.json.JavalinJackson;
 
 public class ModelController implements CrudHandler {
 
+	private static final Logger LOG = Logger.getLogger(ModelController.class);
+
 	@Inject
 	private ResourceManager resourceManager;
 	@Inject
 	private ServerConfiguration serverConfiguration;
 
-	private ResourceSet resourceSet= new ResourceSetImpl();
+	private ResourceSet resourceSet = new ResourceSetImpl();
+	private Map<URI, EObject> modelRepository = new HashMap<URI, EObject>();
```

Comment:
Using a repository is a good idea: I think it could be encapsulated into its own class and use the `ResourceManager` such that the persistence layer is encapsulated as well. This involves the following steps:

* Define an repository API for the `ModelRepository` with CRUD methods
* Remove dependencies to `ResourceManager` and `ResourceSet` and move bits accordingly, e.g. `loadModel`
* Bind `ModelRepositoy` via Guice and inject it instead of the `ResourceManager`

Reply:
Agreed!